R/cnd-message.R

```diff
@@ -1,121 +1,126 @@
-#' Build an error message from a main issue and bullet messages
+#' Build an error message from parts
 #'
 #' @description
 #'
-#' \Sexpr[results=rd, stage=render]{rlang:::lifecycle("experimental")}
-#'
-#' `cnd_message()` assembles an error message from two components:
+#' `cnd_message()` assembles an error message from three generics:
 #'
-#' - The `cnd_issue()` generic. Methods should return a single line.
+#' - `cnd_header()`
+#' - `cnd_body()`
+#' - `cnd_footer()`
 #'
-#' - The `cnd_bullets()` generic. Methods should return a named vector
-#'   of lines. These lines are automatically prefixed with a bullet by
-#'   `cnd_message()` (see the section on error statements).
+#' The default method for the error header returns the `message` field
+#' of the condition object. The default methods for the body and
+#' footer return empty character vectors.
```

Comment:
Oh, I thought the default footer inserted the ``Run `rlang::last_error()`...`

Reply:
I think that's a good idea but we should try it after the CRAN release because changing the mechanism of how this reminder is displayed is a bit tricky (what happens in knitted document etc).